### ISSUE TITLE
inform, warn and abort could pass additional parameters on

### ISSUE THREAD

**Issue by m-stanley** [tidyverse/rlang#245]
Hey - firstly, I'm really enjoying using rlang's easy condition functions.

It would be great though if the convenience functions `inform`, `warn` and `abort` could use the dots to pass on extra arguments to the `cnd_xxx` function that they call.

The reason for this is that, in the same way that these functions already use the type argument to solve the common use case "I want to raise a condition but give it some custom types", there is a second common use case of "I want to raise a condition but to include some extra information inside the condition".

Sometimes it makes sense to build a whole new function for creating those extra-information conditions, but sometimes it doesn't - and in the second case, being able to create them on the fly would be very handy.

At present you must use this sort of construct:

```
extra_info_error <- function(extra, info, variables) {
  msg <-  sprintf("some useful message", extra, info)
  rlang::cnd_error("extra_info",
    extra = extra,
    info = info,
    variables = variables,
    .msg = msg
  )
}
```
Then in your function, calling:

```
stop(extra_info_error(extra = a, info = b, variables = c)
```

With this change, you could use:
```
rlang::abort(sprintf(msg = "some useful message", a, b), type = "extra_info", extra = a, info =b, variables = c)
```
So, if you don't intend to use that sort of condition very frequently, it saves you making a function specifically for it.

**Comment by lionel-** [comment]
I agree, we should start supplying more metadata with errors and we need a proper UI for this.
